apps/backend/src/app/api/latest/auth/oauth/callback/[provider_id]/route.tsx

```diff
@@ -342,19 +359,21 @@ const handler = createSmartRouteHandler({
                               }
                             }
                           });
-                        }
 
-                        await storeTokens();
-                        return {
-                          id: existingUser.projectUserId,
-                          newUser: false,
-                          afterCallbackRedirectUrl,
-                        };
-                      } else if (mergeOAuthMethods === 'RAISE_ERROR') {
-                        // Prevent sign-in with multiple providers
-                        throw new KnownErrors.OAuthConnectionAlreadyConnectedToAnotherUser();
+                          await storeTokens();
+                          return {
+                            id: existingUser.projectUserId,
+                            newUser: false,
+                            afterCallbackRedirectUrl,
+                          };
+                        }
+                        case 'RAISE_ERROR': {
+                          throw new KnownErrors.OAuthConnectionAlreadyConnectedToAnotherUser();
```

Comment:
This error name is wrong. The OAuth connection is not connected to another user! There is another user who already has the same email. The OAuth connection is new (otherwise it would've logged into that OAuth account before)